pip/index.py

```diff
@@ -330,13 +335,23 @@ def mkurl_pypi_url(url):
                 ) or req.url_name
 
         if url_name is not None:
-            locations = [
-                mkurl_pypi_url(url)
-                for url in self.index_urls] + self.find_links
-        else:
-            locations = list(self.find_links)
+            return [mkurl_pypi_url(url) for url in self.index_urls]
+        return []
+
+    def _find_all_versions(self, req):
+        """Find all available versions for req.project_name
+
+        This checks index_urls, find_links and dependency_links
+        All versions are returned regardless of req.specifier
+
+        Cf _link_package_versions for details on which files are accepted
```

Comment:
"Cf" is confusing here, but I won't block a merge on that .. how bout just "see" in the future